lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -148,12 +150,16 @@ bool _hasNonComparableOperands(TypeSystem typeSystem, BinaryExpression node) {
 bool _isCoreInt(DartType type) => type.isDartCoreInt;
 
 bool _isFixNumIntX(DartType type) {
-  if (type is! InterfaceType) {
-    return false;
-  }
+  // todo(pq): add tests that ensure this predicate works with fixnum >= 1.1.0-dev
+  // See: https://github.com/dart-lang/linter/issues/3868
+  if (type is! InterfaceType) return false;
   Element element = type.element;
-  return (element.name == 'Int32' || element.name == 'Int64') &&
-      element.library?.name == 'fixnum';
+  if (element.name != 'Int32' && element.name != 'Int64') return false;
+  var libraryElement = element.library;
+  if (libraryElement is! LibraryElementImpl) return false;
+  var reference = libraryElement.reference;
+  if (reference == null) return false;
+  return reference.name.startsWith('package:fixnum/');
```

Comment:
There aren't any tests. How confident are you that the logic here is correct?

In particular, `Reference.name` is documented to return "The simple name of the reference in its [parent].", which doesn't sound like it's the URI of the library, but maybe.

Reply:
Right.  I've tested this locally with `fixnum` checked out from head.  And verified that it works for the previous `fixnum`.  It's just tricky to test both package structures.

As for digging into the reference, you're 100% right.  Chatting with Konstantin, I was reminded that the source URI is the ticket.